packages/dx-react-grid/src/plugins/table-header-row.jsx

```diff
@@ -32,6 +33,12 @@ export class TableHeaderRow extends React.PureComponent {
             const sorting = getter('sorting');
             const columns = getter('columns');
             const grouping = getter('grouping');
+            const sortingScope = getter('tableColumns')
+              .filter(tableColumn => tableColumn.type === TABLE_DATA_TYPE
+                && (!grouping
+                || !grouping.find(group => group.columnName === tableColumn.column.name)),
```

Comment:
Can we make this logic simple? I mean apply an additional filter on sorting scope if grouping is present.